File: alpinos/alpinos_development/web_form/employee_onboarding_details/employee_onboarding_details.py
import frappe

def get_context(context):
	"""Populate the web-form context from the `onboarding` URL parameter."""
	# Frappe reserves `name` for doc edit/view, so the param is `onboarding`.
	employee_onboarding_name = frappe.form_dict.get('onboarding')
	
	if employee_onboarding_name:
		context.employee_onboarding_name = employee_onboarding_name
		
		if frappe.db.exists("Employee Onboarding", employee_onboarding_name):
			onboarding_doc = frappe.get_doc("Employee Onboarding", employee_onboarding_name)
			
			if onboarding_doc.get('webform_submitted'):
				context.already_submitted = True
				context.submitted_on = onboarding_doc.get('webform_submitted_on')
			else:
				context.already_submitted = False
		else:
			context.invalid_reference = True
	else:
		context.missing_reference = True

File: alpinos/alpinos_development/web_form/employee_onboarding_details/employee_onboarding_details.js
frappe.ready(function() {
	// don't use `name` — Frappe reserves it for doc edit/view
	const urlParams = new URLSearchParams(window.location.search);
	const employeeOnboardingName = urlParams.get('onboarding');

	if (employeeOnboardingName) {
		const hiddenField = document.querySelector('[data-fieldname="employee_onboarding_name"]');
		if (hiddenField) {
			hiddenField.value = employeeOnboardingName;
		}

		if (window.frappe && window.frappe.web_form) {
			window.frappe.web_form.set_value('employee_onboarding_name', employeeOnboardingName);
		}
	} else {
		frappe.msgprint({
			title: __('Error'),
			message: __('Employee Onboarding reference is missing. Please use the link provided in your email.'),
			indicator: 'red'
		});
	}

	if (frappe.web_form && frappe.web_form.context && frappe.web_form.context.already_submitted) {
		frappe.msgprint({
			title: __('Already Submitted'),
			message: __('This form has already been submitted. Please contact HR if you need to make changes.'),
			indicator: 'orange'
		});

		const submitButton = document.querySelector('button[type="submit"]');
		if (submitButton) {
			submitButton.disabled = true;
		}
	}
});

